File: py-cash/cash.py
from cs50 import get_float

def main():
    change = float(get_float("Change: "))
    while change < 0:
        change = float(get_float("Change: "))
    quarters = quarters_charges(change)
    change = round(change - quarters * .25, 2)
    dimes = dimes_charges(change)
    change = round(change - dimes * .10, 2)
    nickels = nickels_charges(change)
    change = round(change - nickels * .05, 2)
    pennies = pennies_charges(change)
    change = round(change - pennies * .01, 2)
    print(quarters + dimes + nickels + pennies)

def quarters_charges(change):
    q = 0
    while change >= .25:
        change -= .25
        q += 1
    return q
def dimes_charges(change):
    d = 0
    while change >= .10:
        change -= .10
        d += 1
    return d
def nickels_charges(change):
    n = 0
    while change >= .05:
        change -= .05
        n += 1
    return n
def pennies_charges(change):
    p = 0
    while change >= .01:
        change -= .01
        p += 1
    return p

main()

File: c-cash/cash.c
#include <cs50.h>
#include <stdio.h>

int quarters_charges(int cents);
int dimes_charges(int cents);
int nickels_charges(int cents);
int pennies_charges(int cents);

int main(void)
{
    // Prompt user for a charges
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    } while (cents < 0);
    int quarters = quarters_charges(cents);
    cents -= quarters * 25; // Update the charges after calculating quarters
    int dimes = dimes_charges(cents);
    cents -= dimes * 10; // Update the charges after calculating dimes
    int nickels = nickels_charges(cents);
    cents -= nickels * 5; // Update the charges after calculating nickels
    int pennies = pennies_charges(cents);
    cents -= pennies * 1; // Update the charges after calculating pennies
    // Print the charges
    printf("The charges are: %iQuarters, %iDimes,  %iNickels, %iPeenies\n", quarters, dimes, nickels, pennies);
    
}

int quarters_charges(int cents)
{
    int quarters = 0;
    while(cents >= 25)
    {
        cents -= 25;
        quarters++;
    }
    return quarters;
}

int dimes_charges(int cents)
{
    int dimes = 0;
    while(cents >= 10)
    {
        cents -= 10;
        dimes++;
    }
    return dimes;
}

int nickels_charges(int cents)
{
    int nickels = 0;
    while(cents >= 5)
    {
        cents -= 5;
        nickels++;
    }
    return nickels;
}

int pennies_charges(int cents)
{
    int pennies = 0;
    while(cents >= 1)
    {
        cents -= 1;
        pennies++;
    }
    return pennies;
}
